src/ae.c

```diff
@@ -484,14 +484,24 @@ int aeWait(int fd, int mask, long long milliseconds) {
     if (mask & AE_READABLE) pfd.events |= POLLIN;
     if (mask & AE_WRITABLE) pfd.events |= POLLOUT;
 
-    if ((retval = poll(&pfd, 1, milliseconds))== 1) {
-        if (pfd.revents & POLLIN) retmask |= AE_READABLE;
-        if (pfd.revents & POLLOUT) retmask |= AE_WRITABLE;
-        if (pfd.revents & POLLERR) retmask |= AE_WRITABLE;
-        if (pfd.revents & POLLHUP) retmask |= AE_WRITABLE;
-        return retmask;
-    } else {
-        return retval;
+    monotime begin = getMonotonicUs();
+    monotime now = begin;
+    while (1) {
+        long long left = milliseconds - (long long)(now - begin) / 1000;
```

Comment:
just to be on the safe side maybe we should check left is not negative (infinite timeout)?
Also what about the case I would like to do a sync operation without timing out?

Reply:
Thanks! I'll fix edge case